packages/shared/components/atoms/AmountBox.svelte

```diff
@@ -1,23 +1,79 @@
 <script lang="typescript">
     import { IPersistedAsset } from '@core/wallet'
-    import { AssetIcon, Text, FontWeight, TextType } from 'shared/components'
+    import { getDecimalSeparator, getGroupSeparator } from '@lib/currency'
+    import { AssetIcon, Text, FontWeight, TextType, Tooltip } from 'shared/components'
 
     export let asset: IPersistedAsset
     export let unit: string
-    export let amount: string
+    export let amount: string = ''
     export let fiatAmount: string = ''
+
+    let displayedAmount: string[] = [amount]
+    let tokenAmountElement: HTMLElement = null
+    let isTooltipVisible = false
+
+    $: {
+        if (amount.length > 12) {
+            displayedAmount = breakAmountIntoLines(amount)
+        } else {
+            displayedAmount = [amount]
+        }
+    }
+
+    function showTooltip(): void {
+        const hasDecimal = amount.includes(getDecimalSeparator())
+        const hasMoreThanTwoDecimalNumbers = hasDecimal && amount.split(getDecimalSeparator())[1].length > 2
+        if (amount.length > 12 && hasDecimal && hasMoreThanTwoDecimalNumbers) {
+            isTooltipVisible = true
+        }
+    }
+
+    function hideTooltip(): void {
+        isTooltipVisible = false
+    }
+
+    function breakAmountIntoLines(amount: string): string[] {
+        const NUMBERS_PER_THOUSAND_GROUP = 3
+        const LINES = 2
+
+        const amountWithTwoDecimalNumbers = amount.slice(0, amount.indexOf(getDecimalSeparator()) + 3)
+        const amountLengthWithoutSeparator = amountWithTwoDecimalNumbers.split(',').join('').length
+        const thousandSeparatorsIndexes: number[] = Array.from(amountWithTwoDecimalNumbers).reduce(
+            (acc, char, i) => (char === getGroupSeparator() ? [...acc, i] : acc),
+            []
+        )
+        const thousandGroups = Math.ceil(amountLengthWithoutSeparator / NUMBERS_PER_THOUSAND_GROUP / LINES)
+
+        const result: string[] = []
+        result.push(amountWithTwoDecimalNumbers.substring(0, thousandSeparatorsIndexes[thousandGroups - 1] + 1))
+        result.push(amountWithTwoDecimalNumbers.substring(result[0].length, amountWithTwoDecimalNumbers.length))
```

Comment:
i think you only take care of numbers with a lot of decimal places. If i have an integer which includes a lot of digits, it doesnt break up the string

Reply:
Fixed it by adding a new algorithm